aten/src/ATen/core/boxing/impl/make_boxed_from_unboxed_functor.h

```diff
@@ -321,18 +370,27 @@ namespace impl {
      * Even though usually dangerous, this is ok here because temporaries live until the end of the statement.
      * TODO We should remove reference_cast once kernels don't take "Tensor&" arguments anymore
      */
-    using ArgTypes = typename guts::infer_function_traits_t<Functor>::parameter_types;
-    return (*functor)(reference_cast<guts::typelist::element_t<ivalue_arg_indices, ArgTypes>>(
+    // We're explicitly filtering out DispatchKeySet from the argument list.
+    // Some kernels take a DispatchKeySet as their first argument in order to plumb keys through the dispatcher.
+    // We don't want to expose the DispatchKeySet type to jit, so we don't include this argument on the stack.
+    // See Note [Plumbing Keys Through The Dispatcher] for the background.
+    using ArgTypes = typename c10::remove_DispatchKeySet_arg_from_func<Functor>::parameter_types;
```

Comment:
I pray we never face the day when someone tries to add DispatchKeySet as a supported JIT schema type and then this does something very strange XD XD XD

Reply:
Oh, this is relevant to your comment:

> And also, I debated wether it would be cleaner/clearer to write a struct that says "filter out the first argument only if it's a DispatchKeySet", rather than "filter out all DispatchKeySet arguments", since that's what the intention really is.

IMO, this is kind of hacky (stripping only the first argument is better), but I think it is relatively unlikely to explode so I'm OK with it, if it helps you avoid spending another day in metaprogramming wasteland.

Reply:
> I pray we never face the day when someone tries to add DispatchKeySet as a supported JIT schema type and then this does something very strange XD XD XD

As unlikely (I hope) as that is, it probably deserves a comment directly in `DispatchKeySet.h` for sanity (I'll add one in)